packages/transformers/js/src/JSTransformer.js

```diff
@@ -75,29 +68,31 @@ export default new Transformer({
     }
 
     asset.type = 'js';
-    if (!asset.ast) {
+    let ast = await asset.getAST();
+    if (!ast) {
```

Comment:
- Generally, when would getAST return a nullish value? Should this be an assertion?
- We should probably add the checks that are in `parse` here well?

Reply:
The Babel Transformer itself doesn't parse at all, instead using the ast if it's available from a prior transformer and sending the code as a string through babel [0]

[0] https://github.com/parcel-bundler/parcel/blob/e75bad2ff98eea4a02290bd74d71d1f77131e68e/packages/transformers/babel/src/babel7.js#L57